src/functions/blockchainFunctions.js

```diff
@@ -158,3 +158,26 @@ function topHoldersPercentOfTotalSupply_ (slug, from, to, numberOfHolders) {
     ]
   }))
 }
+
+function historicalBalanceDedup_ (slug, from, to, address) {
+  var results = getApiClient_().fetchHistoricalBalance(slug, from, to, address)
+  assertHasData_(results)
+
+  var headers = ['Date', 'Balance']
+
+  var resultsArray = results.map(function (result) {
+    return [
+      formatDatetimeField_(result.datetime),
+      formatNumber_(result.balance)
+    ]
+  })
+
+  var deduped = []
+  resultsArray.forEach(function (el, index) {
+    if (deduped.length === 0 || resultsArray.length - 1 === index || deduped[deduped.length - 1][1] !== el[1]) {
```

Comment:
Why is all that needed? The code I shared works:
```js
let deduped = []
arr.forEach(function(el){  if(deduped[deduped.length - 1] != el) deduped.push(el) })
```

Reply:
Hm.. or it's needed so the `...[1]` does not raise

Reply:
Why there are duplicates in the result?

Reply:
The data is returned for every `interval`. If the balance does not change it has the same value as the previous interval. The feature requested (I think by @tzanko-matev) was to be able to fetch only the balances when they change

Reply:
Isn't it possible to make the deduplication with a function in the Sheets?

Reply:
After a little digging, there is actually a function =UNIQUE(range) which deduplicates a row

Reply:
But it won't deduplicate the datetimes next to the results, at least, that won't be so easy. It should be easy for the users to get the deduplicated data IMO and this function does it well.

Reply:
Note that we do not seek unique but removing only the equal neighbors like this [1,2,2,1,1,2] => [1,2,1,2], I am not sure if such spreadsheet function is available out of the box